### PR TITLE
PWX-29351: migrate from k8s.gcr.io to registry.k8s.io

### PR DESCRIPTION
**Pull request by haibinxie**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: drivers/storage/portworx/util/csi_generator.go [hunk 1]
```diff
@@ -260,9 +260,9 @@ func (c *CSIConfiguration) DriverBasePath() string {
 }
 
 func (g *CSIGenerator) getSidecarContainerVersionsV1_0() *CSIImages {
-	provisionerImage := "k8s.gcr.io/sig-storage/csi-provisioner:v3.3.0"
-	snapshotterImage := "k8s.gcr.io/sig-storage/csi-snapshotter:v6.1.0"
-	snapshotControllerImage := "k8s.gcr.io/sig-storage/snapshot-controller:v6.1.0"
+	provisionerImage := "registry.k8s.io/sig-storage/csi-provisioner:v3.3.0"
```
Comment (Pure-AdamuKaapan) [inline drivers/storage/portworx/util/csi_generator.go:263]: Should this use the same constant as above? Replace with a `fmt.Sprintf(...`?

File: drivers/storage/portworx/util/csi_generator.go [hunk 2]
```diff
@@ -272,8 +272,8 @@ func (g *CSIGenerator) getSidecarContainerVersionsV1_0() *CSIImages {
 	// For k8s 1.19 and earlier, use older versions
 	if g.kubeVersion.LessThan(k8sVer1_20) {
 		provisionerImage = "docker.io/openstorage/csi-provisioner:v2.2.2-1"
-		snapshotterImage = "k8s.gcr.io/sig-storage/csi-snapshotter:v3.0.3"
-		snapshotControllerImage = "k8s.gcr.io/sig-storage/snapshot-controller:v3.0.3"
+		snapshotterImage = "registry.k8s.io/sig-storage/csi-snapshotter:v3.0.3"
```
Comment (Pure-AdamuKaapan) [inline drivers/storage/portworx/util/csi_generator.go:275]: Same here, use a constant?

File: drivers/storage/portworx/util/csi_generator.go [hunk 3]
```diff
@@ -284,12 +284,12 @@ func (g *CSIGenerator) getSidecarContainerVersionsV1_0() *CSIImages {
 
 	return &CSIImages{
 		Attacher:                "docker.io/openstorage/csi-attacher:v1.2.1-1",
-		NodeRegistrar:           "k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.6.2",
+		NodeRegistrar:           "registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.6.2",
```
Comment (Pure-AdamuKaapan) [inline drivers/storage/portworx/util/csi_generator.go:287]: And here

File: drivers/storage/portworx/util/util.go [hunk 4]
```diff
@@ -213,7 +213,7 @@ const (
 	// ErrMsgGrpcConnection error message if failed to connect to GRPC server
 	ErrMsgGrpcConnection = "error connecting to GRPC server"
 	// ImageNamePause is the container image to use for the pause container
-	ImageNamePause = "k8s.gcr.io/pause:3.1"
+	ImageNamePause = "registry.k8s.io/pause:3.1"
```
Comment (Pure-AdamuKaapan) [inline drivers/storage/portworx/util/util.go:216]: Use constant?

**Review by Pure-AdamuKaapan — COMMENTED** [review]
A couple places to change. Thanks much for doing this!

File: drivers/storage/portworx/util/csi_generator.go (hunk 1 above)
Comment (haibinxie) [reply drivers/storage/portworx/util/csi_generator.go:263]: sure, i can make the optimization, was just doing a string replacement in the code.

**Review by pure-jliao — APPROVED** [review]
LGTM, only question is for custom registry, why do we keep `k8s.gcr.io` in the path but not this new one.

**Comment by haibinxie** [comment]
> LGTM, only question is for custom registry, why do we keep `k8s.gcr.io` in the path but not this new one.

k8s.gcr.io in the path was a bug, after operator was released we cannot change it anymore due to customer already using it on airgapped env. for the new path, we will change the airgapped script and operator to both not having it in path.

**Pull request merged**